frontend/packages/kubevirt-plugin/src/topology/kubevirt-data-transformer.ts

```diff
@@ -109,3 +116,55 @@ export const getKubevirtTopologyDataModel = (
 
   return Promise.resolve(vmsDataModel);
 };
+
+export const useKubevirtResources: ExtensionHook<WatchK8sResources<any>> = (namespace: string) => {
+  const resources = React.useMemo(
+    () => ({
+      virtualmachines: {
+        isList: true,
+        kind: kubevirtReferenceForModel(models.VirtualMachineModel),
+        namespace,
+        optional: true,
+      },
+      virtualmachineinstances: {
+        isList: true,
+        kind: kubevirtReferenceForModel(models.VirtualMachineInstanceModel),
+        namespace,
+        optional: true,
+      },
+      virtualmachinetemplates: {
+        isList: true,
+        kind: TemplateModel.kind,
+        selector: {
+          matchLabels: { 'template.kubevirt.io/type': 'base' },
+        },
+        optional: true,
+      },
+      migrations: {
+        isList: true,
+        kind: kubevirtReferenceForModel(models.VirtualMachineInstanceMigrationModel),
+        namespace,
+        optional: true,
+      },
+      dataVolumes: {
+        isList: true,
+        optional: true,
+        kind: kubevirtReferenceForModel(models.DataVolumeModel),
+      },
+      vmImports: {
+        isList: true,
+        optional: true,
+        kind: kubevirtReferenceForModel(models.VirtualMachineImportModel),
```

Comment:
don't we need a namespace prop for these resources?

Reply:
I had followed the old function from https://github.com/openshift/console/pull/9355 which didn't have namespace for some resources, added to all now.